core/trino-main/src/main/java/io/trino/server/HttpRequestSessionContextFactory.java

```diff
@@ -218,6 +218,7 @@ public Identity extractAuthorizedIdentity(Optional<Identity> optionalAuthenticat
                 // load enabled roles for authenticated identity, so impersonation permissions can be assigned to roles
                 authenticatedIdentity = Identity.from(authenticatedIdentity)
```

Comment:
I was under impression that this method copies groups.

Reply:
I don't think `authenticatedIdentity` has any groups. Authenticators should not deal with groups.